internal/storage/types/ssblob/ssblob.go

```diff
@@ -0,0 +1,135 @@
+// Package ssblob provides a blob implementation for the Archivematica Storage
+// Service. Use OpenBucket to construct a *blob.Bucket.
+package ssblob
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+
+	"gocloud.dev/blob"
+	"gocloud.dev/blob/driver"
+	"gocloud.dev/gcerrors"
+)
+
+type NotImplemented error
+
+var errNotImpl NotImplemented
+
+type bucket struct {
+	Options
+}
+
+type Options struct {
+	Context context.Context
+	URL     string `json:"string"`
+	Base    string
+}
+
+func openBucket(opts *Options) (driver.Bucket, error) {
+	return &bucket{
+		*opts,
+	}, nil
+}
+
+func OpenBucket(opts *Options) (*blob.Bucket, error) {
+	drv, err := openBucket(opts)
+	if err != nil {
+		return nil, err
+	}
+	return blob.NewBucket(drv), nil
+}
+
+func (b *bucket) ErrorCode(error) gcerrors.ErrorCode {
+	return gcerrors.Unknown
+}
+
+func (b *bucket) As(i interface{}) bool {
+	return false
+}
+
+func (b *bucket) ErrorAs(error, interface{}) bool {
+	return false
+}
+
+func (b *bucket) Attributes(ctx context.Context, key string) (*driver.Attributes, error) {
+	return nil, errNotImpl
+}
+
+func (b *bucket) ListPaged(ctx context.Context, opts *driver.ListOptions) (*driver.ListPage, error) {
+	return nil, errNotImpl
+}
+
+func (b *bucket) NewRangeReader(ctx context.Context, key string, offset, length int64, opts *driver.ReaderOptions) (driver.Reader, error) {
+	client := http.Client{}
```

Comment:
I've suggested earlier to make the HTTP client a member of the bucket struct so here we would just be using `b.client` instead of having to create a new client each time the user wants to read. http.Client is thread-safe and best shared in a struct.